icu4c/source/common/unicode/uset.h

```diff
@@ -1700,9 +1700,10 @@ class USetStringIterator {
  * using U_HEADER_NESTED_NAMESPACE::USetStrings;
  * LocalUSetPointer uset(uset_openPattern(u"[abcçカ🚴{}{abc}{de}]", -1, &errorCode));
  * for (auto s : USetStrings(uset.getAlias())) {
- *     UnicodeString us(s);
- *     std::string u8;
- *     printf("uset.string length %ld \"%s\"\n", (long)s.length(), us.toUTF8String(u8).c_str());
+ *     int32_t len32 = s.length();
+ *     char utf8[200];
+ *     u_strToUTF8WithSub(utf8, 199, nullptr, s.data(), len32, 0xFFFD, nullptr, errorCode);
```

Comment:
Considering how many of these Frank has fixed just in the past few years, to correct off-by-one errors and hardcoded lengths forgotten when buffere sizes were changed, I think it's a really good idea to use `UPRV_LENGTHOF` or `sizeof` expressions even in commented-out sample code.